substrafl/algorithms/pytorch/torch_fed_pca_algo.py

```diff
@@ -302,7 +302,7 @@ def predict(self, datasamples: Any, shared_state: Any = None, predictions_path:
         # Create torch dataset
         predict_dataset = self._dataset(datasamples, is_inference=True)
 
-        dataloader_batchsize = self._batch_size or len(predict_dataset)
+        dataloader_batchsize = min(self._batch_size, len(predict_dataset)) if self._batch_size else len(predict_dataset)
```

Comment:
It's the only two algorithms that have a batch_size?

Reply:
They are the only one that use a given batch size in the algo, due to the fact that they are "epoch based".  the others algo use the batch size of the index generator.